File: pyround1/492.构造矩形.py
#
# @lc app=leetcode.cn id=492 lang=python
#
# [492] 构造矩形
#

# @lc code=start
class Solution(object):
    def constructRectangle(self, area):
        """
        :type area: int
        :rtype: List[int]
        """
        W = int(math.sqrt(area))
        while area % W != 0:
            W -= 1
        return [area/W, W]
        
# @lc code=end

File: 492.构造矩形.cpp
/*
 * @lc app=leetcode.cn id=492 lang=cpp
 *
 * [492] 构造矩形
 */

// @lc code=start
class Solution {
public:
    vector<int> constructRectangle(int area) {
        int sub = sqrt(area);
        while(area % sub != 0)
            sub--;
        return {area/sub, sub};

    }
};
// @lc code=end

